### PR TITLE
fix state at end of completed video

### PR DESCRIPTION
**Pull request by joelhooks**
image

### CODE REVIEW

File: src/components/pages/lessons/overlay/course-pitch-overlay.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import Link from 'next/link'
+import Image from 'next/image'
+import * as React from 'react'
+import {track} from 'utils/analytics'
+import noop from 'utils/noop'
+
+const CoursePitchOverlay: React.FunctionComponent<{
```
Comment (joelhooks) [inline src/components/pages/lessons/overlay/course-pitch-overlay.tsx:7]: direct cut and paste from the next lesson overlay

File: src/pages/lessons/[slug].tsx [hunk 2]
```diff
@@ -136,8 +137,25 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
...
+  const [isIncomingAnonViewer, setIsIncomingAnonViewer] =
+    React.useState<boolean>(false)
+
+  React.useEffect(() => {
+    const storage = window?.sessionStorage
+    if (!storage) return
+
+    const prevPath = storage.getItem('prevPath')
+
+    if (!prevPath && !subscriber && !viewer) {
```
Comment (joelhooks) [inline src/pages/lessons/[slug].tsx:152]: if there is no previous path logged then we know they just dropped into this lesson and they aren't anon if we have a subscriber or viewer

File: src/machines/lesson-player-machine.ts [hunk 3]
```diff
@@ -27,12 +28,13 @@ export type PlayerStateEvent =
   | {type: 'PLAY'}
   | {type: 'PAUSE'}
   | {type: 'LOADED'}
-  | {type: 'LOADED'; lesson: any; viewer: any}
+  | {type: 'LOADED'; lesson: any; viewer: any; isIncomingAnonViewer?: boolean}
```
Comment (jbranchaud) [inline src/machines/lesson-player-machine.ts:31]: You added `isIncomingAnonViewer` to the type, but it isn't set or used within the machine. It looks like the lesson page gets by fine with it as component local state, so I think you can ditch the type change here.

Comment (jbranchaud) [inline src/machines/lesson-player-machine.ts:31]:
```suggestion
  | {type: 'LOADED'; lesson: any; viewer: any}
```

**Pull request merged**